### DISCUSSION TITLE
RX-A870 presets

### DISCUSSION THREAD

**Discussion by duusiek** [mvdwetering/yamaha_ynca#291] [General]
Hi All,
I used since long time integration MusicCast which worked quite well. Since some HAOS update it significantly slowwed down. When I turned ON hardware woked up but for reaction in HA I have to wait some seconds. Rather 20 seconds than 2. Then I can select some source or do anything. Normally I can select NET Radio and explore "library" to choose favorite station.

I try to configure averything again. I found also to integrate my RX-A870 as Yamaha (YNCA). All works much faster - I see HA reaction in a while when I pres anything. Unfortunatelly when I try to explore Favorities like beffore button "explore media" goes to Presets.

My questions (1&2 or 1or2):
1. How to configure presets to assign my favorities radio station to each presets? (Yamaha (YNCA) integration)
2. How to improove responsiveness of MusicCast integration?

![2024-06-19_12h50_02](https://github.com/mvdwetering/yamaha_ynca/assets/59575372/ef7c04d1-9ab8-4542-a09f-76a83c9fac1d)

![2024-06-19_12h50_23](https://github.com/mvdwetering/yamaha_ynca/assets/59575372/9082a708-5232-4d24-a4f8-084344b2852e)

**Comment by mvdwetering** [comment]
I have not found a way yet to have a real mediabrowser like in your screenshot in a reliable/stable way. There is an open issue for that (#213 ), but unlikely to get implemented soon, so until that time you will have to use the presets.

Presets can be stored by using the `store_preset` service that this integration offers. It stores the current playing "thing" under the preset with the number provided. So you start netradio on the station you like and then call `store_preset` with the preset number you want to store it under. You will have to remember which preset contains which station. I can not get that information from the receiver.
It is explained in a bit more detail in the [Presets section of the Readme](https://github.com/mvdwetering/yamaha_ynca?tab=readme-ov-file#presets)

I can't help with your MusicCast question as my receiver does not support it and I know nothing about it. You should probably submit an issue with Home Assistant about the slow response in the MusicCast integration.
